### PR TITLE
Support `KFAnimatedImage` and `AnimatedImageView` on macOS

### PR DESCRIPTION
**Pull request by yeatse**
So that we can play animated GIFs on SwiftUI for macOS

### CODE REVIEW

**Comment by onevcat** [comment]
Perfect at the first glance! I will look into them shortly when having a chance. Thank you!

File: Sources/SwiftUI/KFAnimatedImage.swift [hunk 1]
```diff
@@ -75,6 +76,35 @@ public struct KFAnimatedImageViewRepresenter: UIViewRepresentable, KFImageHoldin
         uiView.image = image
     }
 }
+#else
+@available(macOS 11.0, *)
+public struct KFAnimatedImageViewRepresenter: NSViewRepresentable, KFImageHoldingView {
```
Comment (onevcat) [inline Sources/SwiftUI/KFAnimatedImage.swift:81]: I see quite a lot of duplicated code.

Maybe we can combine these two `KFAnimatedImageViewRepresenter`s into one? Something like:

```swift
#if !os(macOS)
typealias KFCrossPlatformViewRepresentable = UIViewRepresentable
#else
typealias KFCrossPlatformViewRepresentable = NSViewRepresentable
#endif
```

What your opinion?

File: Sources/Views/AnimatedImageView.swift [hunk 2]
```diff
@@ -162,6 +173,101 @@ open class AnimatedImageView: UIImageView {
     // A flag to avoid invalidating the displayLink on deinit if it was never created, because displayLink is so lazy.
     private var isDisplayLinkInitialized: Bool = false
     
+#if os(macOS)
+    class DisplayLink {
```
Comment (onevcat) [inline Sources/Views/AnimatedImageView.swift:177]: Can we move this class to another new file to keep this file a bit cleaner?

File: Sources/SwiftUI/KFAnimatedImage.swift (hunk 1 above)
Comment (yeatse) [reply Sources/SwiftUI/KFAnimatedImage.swift]: Good idea. Updated

File: Sources/Views/AnimatedImageView.swift (hunk 2 above)
Comment (yeatse) [reply Sources/Views/AnimatedImageView.swift]: Done

**Comment by onevcat** [comment]
@yeatse Perfect! Thanks for the epic efforts!

**Pull request merged**